raiko-guest/src/one_shot.rs

```diff
@@ -0,0 +1,227 @@
+use std::{
+    fs::{self, File, OpenOptions},
+    io::prelude::*,
+    path::PathBuf,
+    str::FromStr,
+};
+
+use anyhow::{anyhow, bail, Error, Result};
+use secp256k1::{hashes::sha256, rand::rngs::OsRng, All, Message, PublicKey, Secp256k1, SecretKey};
+use zeth_lib::{
+    block_builder::{TaikoBlockBuilder, TaikoStrategyBundle},
+    consts::TAIKO_MAINNET_CHAIN_SPEC,
+    host::Init,
+};
+
+use crate::app_args::{GlobalOpts, OneShotArgs};
+
+pub const ATTESTATION_QUOTE_DEVICE_FILE: &str = "/dev/attestation/quote";
+pub const ATTESTATION_TYPE_DEVICE_FILE: &str = "/dev/attestation/attestation_type";
+pub const ATTESTATION_USER_REPORT_DATA_DEVICE_FILE: &str = "/dev/attestation/user_report_data";
+pub const PRIV_KEY_FILENAME: &str = "priv.key";
+
+pub async fn one_shot(global_opts: GlobalOpts, args: OneShotArgs) -> Result<()> {
+    if args.no_sgx {
+        eprintln!("Running without SGX");
+    }
+
+    if !is_bootstrapped(&global_opts.secrets_dir) {
+        bail!("Application was not bootstrapped. Bootstrap it first.")
+    }
+
+    let path_str = args.blocks_data_file.to_string_lossy().to_string();
+    let block_no = u64::from_str(&String::from(
+        args.blocks_data_file
+            .file_prefix()
+            .unwrap()
+            .to_str()
+            .unwrap(),
+    ))
+    .unwrap();
+    eprintln!("Reading input file {} (block no: {})", path_str, block_no);
+
+    // TODO FIXME
+    let block_header_hash_str = String::from("foobar");
+    // let block_header_hash_str = get_data_to_sign(path_str, block_no).await?;
+    eprintln!("Data to be signed: {}", block_header_hash_str);
+
+    let public_key = sgx_sign(global_opts.secrets_dir, block_header_hash_str)?;
+
+    if !args.no_sgx {
+        print_sgx_info(public_key)?
+    }
+
+    Ok(())
+}
+
+pub fn bootstrap(global_opts: GlobalOpts) -> Result<()> {
+    // TODO REMOVEME temporary debug
+    for path in fs::read_dir("/secrets").unwrap() {
+        println!("Entry: {}", path.unwrap().path().display())
+    }
+
+    let (next_private_key, next_public_key, _secp) = generate_new_keypair(global_opts.secrets_dir)?;
+
+    println!("Next private key: 0x{}", next_private_key.display_secret());
```

Comment:
Printing private keys is of course temporary, just to test the key rotation works well.

Reply:
Commented it out.